feat: Add Support for creating user entries with alias flag

elijah-roberts opened the pull request:
*Issue #, if available: Fixes #5164

*Description of changes: Added support for creating user name in kube-config with alias (simliar to the way context works with alias)
Added appropriate testing for user name creation with and without alias flag


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

elijah-roberts commented:
@kyleknap @elysahall  Anything else I can do for this PR?

nckturner commented:
@elijah-roberts in your example from the issue, the context name is set to the same value as the user name.  Is that missing from this PR?

nckturner commented:
Nvm, I think I see how it works now, alias was already provided as a flag to set context name.  While using alias for both username and context name would work, it seems like we could add a `--user` flag and not worry about breaking any users relying on the current behavior.

kdaily commented:
Hi @elijah-roberts, can you address @nckturner's suggestions?

elijah-roberts commented:
> Nvm, I think I see how it works now, alias was already provided as a flag to set context name. While using alias for both username and context name would work, it seems like we could add a `--user` flag and not worry about breaking any users relying on the current behavior.

That would work for me 👍  I can plan on implementing a flag and testing that out.

elijah-roberts commented:
@nckturner @kdaily  I appreciate your reviewing my PR. I updated with the changes you suggested by breaking out this logic into a new flag `user-alias`. I have updated tests, and confirmed that it is working as expected for myself.

elijah-roberts commented:
@nckturner @kdaily  Looks like the codecov/project failed despite the fact that I added testes for the code that I added. Is there something else that I need to do?

kdaily commented:
@nckturner - let me know here if this is good to go!

@elijah-roberts - I'll look into the coverage failure.

Thanks both!

elijah-roberts commented:
Bump... @kdaily  Any update on this?

nckturner commented:
@kdaily this LGTM

potto007 commented:
My team could really use this feature. @kdaily is this going to be merged?

Thanks!

joaoubaldo commented:
+1 waiting for this to be merged. Any updates @kdaily?

kdaily commented:
@joaoubaldo, @potto007, @elijah-roberts - this feature still needs review from the CLI team, and it is on the agenda for review. Marking it as such now. It's on the radar, but I don't know when it will be able to pull in.

joshdk commented:
+1 - This feature would be a welcome addition

elijah-roberts commented:
@joaoubaldo, @potto007, @kdaily bumping again, this is a simple non-breaking change that is useful for supporting the creation of multiple kube-configs for different roles. For enterprise EKS users typically there will be a daily role that has limited permissions, and mappings. There will also be a break-glass type admin role. Current state to support both of these you have to manually manage your kubeconfig file, or constantly overwrite the associated user each time you update the config with a different role. 

Any chance we can get a review on this ? I am happy to make any necessary changes.

elijah-roberts commented:
Pulled in latest changes from develop, and the code coverage report looks good now

yuxiang-zhang commented:
We are trying to prioritize this feature request and get it merged soon. Note that this implements the same feature as #5413

joshuamkite commented:
> We are trying to prioritize this feature request and get it merged soon. Note that this implements the same feature as #5413

Thanks but that issue has also been open for the best part of 3 years. What's the holdup?
